pkg/machine/controller/machine/controller.go

```diff
@@ -311,8 +312,10 @@ func (m *Lifecycle) saveConfig(config *machineconfig.MachineConfig, machineDir s
 
 func (m *Lifecycle) isNodeInAppliedSpec(machine *v3.Machine) (bool, error) {
 	cluster, err := m.clusterLister.Get("", machine.Spec.ClusterName)
-	if err != nil {
+	if err != nil && !kerror.IsNotFound(err) {
 		return false, err
+	} else if kerror.IsNotFound(err) {
```

Comment:
@StrongMonkey err can be nil at this point, just wanna make sure kerror.IsNotFound can handle it. Or may be it would be cleaner to define the if clause as:

```
if err != nil {
   if err.IsNotFound(err) {
    return false, nil
}
return false, err
}